tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,130 @@
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(60 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("helm manifest are applied", func() {
+			if skipDeploy == "true" {
+				Skip("Skipping the deployment of the operator and the tests")
+			} else {
+				Eventually(deployOperator).Should(Succeed())
+			}
+
+			It("the operator is running", func() {
+				Eventually(getOperatorState).Should(Equal("Running"))
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio control plane", func() {
+		baseDir := filepath.Dir(filepath.Dir(filepath.Dir(wd)))
+		istioVersions, err := getIstioVersions(filepath.Join(baseDir, "versions.yaml"))
+		if err != nil {
+			Fail(fmt.Sprintf("Error getting istio versions from version.yaml file: %v", err))
+		}
+
+		for _, version := range istioVersions {
+			Context(fmt.Sprintf("for supported version %s", version), func() {
+				When(fmt.Sprintf("the istio control plane is installed with version %s", version), func() {
+					BeforeAll(func() {
+						Eventually(createNamespaceIfNotExists).WithArguments(controlPlaneNamespace).Should(Succeed())
+						Eventually(deployIstioControlPlane).WithArguments(version).Should(Succeed())
+
+						DeferCleanup(func() {
+							Eventually(deleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed())
+							Eventually(namespaceIsDeleted).WithArguments(controlPlaneNamespace).Should(Equal(true))
+						})
+					})
+
+					It("istio resource is ready and running", func() {
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Reconciled").Should(Succeed())
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Ready").Should(Succeed())
+
+						istiodPodName, err := getPodNameFromLabel(controlPlaneNamespace, "app=istiod")
+						if err != nil {
+							Fail("Error getting pod name from deployment")
+						}
+
+						Eventually(getPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"))
+					})
+
+					It("the istio resource version match the installed version", func() {
+						Expect(getInstalledIstioVersion()).Should(Equal(version))
+					})
+
+					It("istio resource stopped reconciling", func() {
+						Eventually(checkIstioStoppedReconciling).WithTimeout(120 * time.Second).Should(Succeed())
+					})
+
+					It("only istiod is deployed", func() {
+						Expect(checkOnlyIstioIsDeployed(controlPlaneNamespace)).To(Equal(true))
```

Comment:
This should also be inlined probably. I don't see us having to call this function elsewhere, since it's so specific.

Reply:
Done